### PR TITLE
feat: planet visit order optimizer

### PR DESCRIPTION
**Pull request by teal-bauer**
Adds orbital position calculation and route optimization for DSS targets, as described in #875.

## Changes

- **OrbitalMechanics.cs** (new): Keplerian orbit solver (Newton-Raphson for eccentric anomaly) + perifocal-to-inertial 3D positioning with recursive parent accumulation. Double-precision vectors because `System.Numerics.Vector3` loses precision at 100+ AU distances. Route optimizer uses exact permutation search for ≤10 targets, nearest-neighbor + 2-opt for larger systems.
- **SystemData.cs**: `getDssRemainingBodyIds()` reuses the existing DSS skip filters. `CalculateOptimalRoute(targetBodyIds)` builds the orbital model and sets `visitOrder` on each body.
- **PlotSysStatus.cs**: Shows visit route as `1 > 3 > 5 > 5a > 4 > 2 > 6` with destination body highlighting. DEBUG builds get a diagnostic overlay with hop distances and route comparison.

Implements #875

## Status

This is a first working version — tested manually in a handful of systems, route looks correct compared to the orrery view. Still needs more testing with edge cases (missing orbital data, barycentre-heavy systems, very large systems with >10 DSS targets).

### CODE REVIEW

**Comment by Jessidhia** [comment]
Out of curiosity, I wonder if they gave the Voyager probe POIs in the Sol system hyperbolic orbits (negative eccentricity). Probably not something the optimizer would actually encounter, though, since Elite otherwise doesn't do comets 😞

It is also unclear what happens to the argument of Periapsis / MeanAnomaly when something has an eccentricity of *exactly* 1, or AscendingNode when the OrbitalInclination is exactly 0, but I don't know what body / system can be used to check for it.

**Comment by teal-bauer** [comment]
Didn't Elite use to do comets? But yeah, I doubt they'd special case for something small like that. Would be interesting to log this data and see if any systems have such borderline values.

I've been doing some more testing and I think this only needs a UX decision before it's ready to merge - right now I show the DSS list and then the visit order underneath - @njthomson Do you think we should just merge that into one, or keep it separate?

**Comment by teal-bauer** [comment]
Example what it looks like rn:
image
image

**Comment by njthomson** [comment]
I am back from my travels ... let me look at the code and ponder the UX design.

File: SrvSurvey/game/SystemData.cs [hunk 1]
```diff
@@ -1910,6 +1940,94 @@ public async Task<double> getNebulaDist()
...
+                    if (body != null)
+                        body.visitOrder = i;
+                }
+
+                lastCalculator = calculator;
+                Game.log($"Calculated optimal route for {targetBodyIds.Count} bodies in {this.name}");
+            }
+            catch (Exception ex)
+            {
+                Game.log($"Error calculating optimal route: {ex.Message}");
```
Comment (njthomson) [inline SrvSurvey/game/SystemData.cs]: May I recommend including StackTrace, so it's automatically written to log files:
```
$"Error calculating optimal route: {ex.Message}\r\n\t{ex.StackTrace}"
```

File: SrvSurvey/game/SystemData.cs
Comment (njthomson) [inline SrvSurvey/game/SystemData.cs]: Ideally (or in a subsequent PR) we would populate orbital mechanic numbers from EDSM.
(Assuming it has all of them?)

**Comment by njthomson** [comment]
@teal-bauer - if you can get the main entry points shielded behind a new setting, I see no reason not to get this in and try it out.

**Comment by teal-bauer** [comment]
Aye! I hope to find some time on the weekend!

**Comment by Fenris159** [comment]
A replacement draft is now open at #1023, based on current main. It preserves @teal-bauer original two commits and authorship, and adds a separate hardening commit for the owner feedback and review findings discussed here. Keeping #884 as the original design and review history; maintainers can decide when to close it as superseded.
